packages/oidc-auth/src/index.ts

```diff
@@ -60,47 +60,84 @@ type OidcAuthEnv = {
   OIDC_COOKIE_DOMAIN?: string
 }
 
+/**
+ * Configure the OIDC variables programmatically.
+ * If used, should be called before any other OIDC middleware or functions for the Hono context.
+ * Unconfigured values will fallback to environment variables.
+ */
+export const initOidcAuthMiddleware = (config: Partial<OidcAuthEnv>) => {
+  return createMiddleware(async (c, next) => {
+    setOidcAuthEnv(c, config)
+    await next()
+  })
+}
+
+/**
+ * Configure the OIDC variables.
+ */
+const setOidcAuthEnv = (c: Context, config?: Partial<OidcAuthEnv>) => {
+  const currentOidcAuthEnv = c.get('oidcAuthEnv')
+  if (currentOidcAuthEnv !== undefined) {
+    throw new HTTPException(500, { message: 'OIDC Auth env is already configured' })
+  }
+  const ev = env<Readonly<OidcAuthEnv>>(c)
```

Comment:
Changed to take a Readonly reference to the environment variables to avoid accidental mutation